### PR TITLE
add hours components/ change hours components in front end

### LINKED ISSUE #53: Add hours field into course details response
Add the `hours` field into the course details response.

This involves updating the scraper library version in the dependencies and adding another "badge" to the frontend code.

### PR DESCRIPTION
**Pull request by chsunwoo1002**
## Overview

Closes #53

### CODE REVIEW

File: ui-debug.log [hunk 1]
```diff
@@ -0,0 +1 @@
+Web / API server started at http://localhost:4000
```
Comment (aomi) [inline ui-debug.log:1]: Delete this file.

File: src/app/content/components/Course.tsx [hunk 2]
```diff
@@ -99,7 +100,9 @@ export function CourseInfo({
         {credits && (
           <CourseShield bg="purple.200" title="Credits">
             <Heading size="md" title="lecture hours per week">
-              {credits}
+              {credits.credits.max === credits.credits.min
+                ? credits.credits.max
+                : `${credits.credits.min} ~ ${credits.credits.max}`}
```
Comment (aomi) [inline src/app/content/components/Course.tsx:105]: Nice!

File: src/app/content/components/Course.tsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { Box, Heading, Text } from '@chakra-ui/layout';
 import { BackgroundProps, Divider, Flex } from '@chakra-ui/react';
 import { PropsWithChildren } from 'react';
+import { createMethodSignature } from 'typescript';
```
Comment (aomi) [inline src/app/content/components/Course.tsx:4]: Not sure how this got here but remove this import.

File: package.json [hunk 4]
```diff
@@ -106,6 +106,7 @@
     "prettier": "^2.2.1",
     "storybook-addon-designs": "^5.4.5"
   },
+  "proxy": "http://localhost:2999",
```
Comment (aomi) [inline package.json:109]: Remove this as it was just a little hack for development.

File: functions/src/server.ts [hunk 5]
```diff
@@ -10,7 +10,7 @@ import * as openapi from '../build/swagger.json';
 
 export const app = express();
 
-const port = process.env.PORT || 3001;
+const port = process.env.PORT || 2999;
```
Comment (aomi) [inline functions/src/server.ts:13]: Change this back to `3001` instead of `2999`.

File: functions/src/courses/Course.model.ts [hunk 6]
```diff
@@ -28,4 +28,8 @@ export interface CourseDetails
    * @example "ELEC260"
    */
   formally?: string;
+  /**
+   * Course hours
+   */
+  hoursCatalogText?: { lecture: string; lab: string; tutorial: string };
```
Comment (aomi) [inline functions/src/courses/Course.model.ts:34]: This isn't needed anymore.

File: functions/src/courses/Course.controller.ts [hunk 7]
```diff
@@ -3,6 +3,8 @@ import { Term } from '../constants';
 import { Course, CourseDetails } from './Course.model';
 import { CoursesService } from './Course.service';
 
+// /courses/202101
+
```
Comment (aomi) [inline functions/src/courses/Course.controller.ts:7]: Remove these lines.

File: firestore-debug.log [hunk 8]
```diff
@@ -0,0 +1,7 @@
+API endpoint: http://localhost:8080
+If you are using a library that supports the FIRESTORE_EMULATOR_HOST environment variable, run:
+
+   export FIRESTORE_EMULATOR_HOST=localhost:8080
+
+Dev App Server is now running.
+
```
Comment (aomi) [inline firestore-debug.log:7]: Remove this file.

**Review by aomi — COMMENTED** [review]
Just some minor edits needed but overall good stuff.

File: functions/src/courses/Course.model.ts (hunk 6 above)
Comment (chsunwoo1002) [reply functions/src/courses/Course.model.ts]: @aomi when I delete this line, /funcitons/build/routes.ts file and functions/build/swagger.json file automatically changed. is it okay?

Comment (aomi) [reply functions/src/courses/Course.model.ts]: Yeah that's expected and what we want :+1:

File: functions/src/server.ts (hunk 5 above)
Comment (aomi) [reply functions/src/server.ts:13]: @chsunwoo1002 ^^

Comment (chsunwoo1002) [reply functions/src/server.ts]: @aomi fixed it!

**Review by aomi — APPROVED** [review]

**Pull request merged**